task/deprecated-image-check/0.4/deprecated-image-check.yaml

```diff
@@ -0,0 +1,154 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: deprecated-image-check
+spec:
+  description: >-
+    Identifies the unmaintained and potentially insecure deprecated base images.
+    Pyxis API collects metadata from image repository, and Conftest applies supplied policy to identify the deprecated images using that metadata.
+  params:
+    - name: POLICY_DIR
+      description: Path to directory containing Conftest policies.
+      default: "/project/repository/"
+    - name: POLICY_NAMESPACE
+      description: Namespace for Conftest policy.
+      default: "required_checks"
+    - name: BASE_IMAGES_DIGESTS
+      description: Digests of base build images.
+      default: ""
+    - name: IMAGE_URL
+      description: Fully qualified image name.
+    - name: IMAGE_DIGEST
+      description: Image digest.
+
+  results:
+    - name: PYXIS_HTTP_CODE
+      description: HTTP code returned by Pyxis API endpoint.
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+
+  steps:
+    - name: check-images
+      image: quay.io/redhat-appstudio/hacbs-test:v1.4.0@sha256:54d49b37c9a2e280d42961a57e4f7a16c171d6b065559f1329b548db85300bea
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      # also per direction from Ralph Bean, we want to use image digest based tags to use a cue to automation like dependabot or renovatebot to periodially submit pull requests that update the digest as new images are released.
+      env:
+        - name: POLICY_DIR
+          value: $(params.POLICY_DIR)
+        - name: POLICY_NAMESPACE
+          value: $(params.POLICY_NAMESPACE)
+        - name: BASE_IMAGES_DIGESTS
+          value: $(params.BASE_IMAGES_DIGESTS)
+        - name: IMAGE_URL
+          value: $(params.IMAGE_URL)
+        - name: IMAGE_DIGEST
+          value: $(params.IMAGE_DIGEST)
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+        source /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        SBOM_FILE_PATH="/tmp/sbom.json"
+        IMAGES_TO_BE_PROCESSED_PATH="/tmp/images_to_be_processed.txt"
+
+        success_counter=0
+        failure_counter=0
+        error_counter=0
+        warnings_counter=0
+
+        # Get base images from SBOM
+        # images are without tag
+        echo "Fething SBOM for image ${IMAGE_URL}@${IMAGE_DIGEST}"
+        cosign download sbom "${IMAGE_URL}@${IMAGE_DIGEST}" > ${SBOM_FILE_PATH}
+        cat "${SBOM_FILE_PATH}" | jq -r '.formulation? // empty | .[] | .components? // empty | .[] | select(.type=="container") | .name' > ${IMAGES_TO_BE_PROCESSED_PATH}
```

Comment:
If you want to select exactly the base images that Konflux would report in BASE_IMAGES_DIGESTS, you can add

```jq
select(
  any((.properties // empty)[];
  .name | test("^konflux:container:is_(base|builder)_image"))
)
```

At the moment it only makes a hypothetical difference: what do you want deprecated-image-check to do if the `.formulation` part of the SBOM contains `container` type components reported by something other than Konflux?

Reply:
updtated, thanks for the hint